### PR TITLE
Improvements for autoComplete, dropDown and button components

### PR DESCRIPTION
**Pull request by AlleeX**
- style improvements for button
- added disabled state, dataAttrs and mods for dropDown
- added autoComplete="off" for autoComplete
- performance optimisation for autocomplete

### CODE REVIEW

File: components/dropDown/dropDown.js [hunk 1]
```diff
@@ -83,29 +83,46 @@ class DropDown extends Component {
...
     const isFiltermode = this.props.filterMode;
-    const mods = {
-      'filter': isFiltermode,
-      'state-active': isFiltermode && this.props.options.some(option => option.checked),
-    };
+
+    filterMode && mods.push('filter');
+    filterMode && this.props.options.some(option => option.checked) && mods.push('state-active');
 
     const className = getClassNamesWithMods('ui-dropdown', mods);
```
Comment (reaktivo) [inline components/dropDown/dropDown.js]: `getClassNamesWithMods` will filter out falsy values by itself.

    const className = getClassNamesWithMods('ui-dropdown', [
      filterMode && 'filter',
      filterMode && options.some(option => option.checked) && 'state-active'
    ]);

File: components/dropDown/dropDown.js [hunk 2]
```diff
@@ -83,29 +83,46 @@ class DropDown extends Component {
...
-        onChange={this.onChange}
-        optionComponent={isFiltermode ? DropdownFilterOptionComponent : undefined}
-        options={this.props.options}
-        placeholder={this.props.placeholder}
-        scrollMenuIntoView={this.props.scrollMenuIntoView}
-        searchable={this.props.searchable}
-        value={this.props.value}
-      />
+      <div {...getDataAttributes(dataAttrs)}>
+        <Select
```
Comment (reaktivo) [inline components/dropDown/dropDown.js]: The Select component can handle data attributes fine, via inputProps. No need for wrapping.

    <Select inputProps={getDataAttributes(dataAttrs)}>

Comment (AlleeX) [inline components/dropDown/dropDown.js]: Unfortunately 'react-select' don't have this functionality

Comment (reaktivo) [inline components/dropDown/dropDown.js]: Hmm, https://github.com/JedWatson/react-select/blob/5f7c4c69e74f1435b4345847dc7e0d824901f5d7/README.md#further-options

Comment (AlleeX) [inline components/dropDown/dropDown.js]: yep, but input renders only for searchable dropDown

File: components/dropDown/dropDown.js (hunk 1 above)
Comment (AlleeX) [inline components/dropDown/dropDown.js]: done. thanks!)

**Pull request merged**